### PR TITLE
[WIP] add some `gdal_utils` support functions

### PR DESCRIPTION
**Pull request by idanmiara**
## What does this PR do?

draft
add some `gdal_utils` support functions

### CODE REVIEW

File: gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/ogr_utils.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        geom = ogr.CreateGeometryFromWkt(wkt)
+        feature.SetGeometry(geom)  # Set the feature geometry
+        layer.CreateFeature(feature)  # Create the feature in the layer
+        feature.Destroy()  # Destroy the feature to free resources
+    # Destroy the data source to free resources
+    ds.Destroy()
+
+
+def ogr_get_layer_extent(lyr: ogr.Layer) -> GeoRectangle:
+    result = None
```
Comment (rouault) [inline gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/ogr_utils.py:58]: why not use Layer.GetExtent() ?

**Review by rouault — COMMENTED** [review]
I'm not sure what to do with this PR. It is not obvious that all those utility functions are needed, as they are not used by anything else.
I'm really wondering why there's a specific need for UTM handling in osr_utm_util.py 
And there are no tests for all the new functions

File: gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/osr_utm_util.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    extent_points = []
+    for x in x_arr:
+        for y in y_arr:
+            extent_points.append((x, y))
+        y_arr.reverse()
+    return extent_points
+
+
+def proj_string_from_utm_zone(zone: Optional[Real] = None, datum_str='+datum=WGS84') -> str:
```
Comment (rouault) [inline gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/osr_utm_util.py:83]: proj string for CRS are a legacy. We should really try hard not to use them

**Comment by idanmiara** [comment]
> I'm not sure what to do with this PR. It is not obvious that all those utility functions are needed, as they are not used by anything else.
> I'm really wondering why there's a specific need for UTM handling in osr_utm_util.py
> And there are no tests for all the new functions

Thanks for your review, I meant to use these in some utils and add tests, so I wanted to test them first, that's why I marked it as [WIP]. I didn't expect to merge it beforehand.

File: gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/ogr_utils.py (hunk 1 above)
Comment (idanmiara) [reply gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/ogr_utils.py:58]: Thanks, I'll look into it.

File: gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/osr_utm_util.py (hunk 2 above)
Comment (idanmiara) [reply gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/osr_utm_util.py:83]: What would be a non-legacy way to define the CRS that sits between `EPSG:326xy` and `EPSG:326xy+1` i.e. "zone xy+0.5" ?

Comment (rouault) [reply gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/osr_utm_util.py:83]: use the SpatialReference.SetTransverseMercator() method with appropriate values

**Comment by idanmiara** [comment]
I'll reopen this PR when it's ready.

**Pull request closed without merging**